Assets/_BForBoss/_Weapons/Scripts/Weapons/Weapon.cs

```diff
@@ -111,9 +113,10 @@ private void Fire()
              OnFireWeapon?.Invoke(_weaponProperties.BulletsPerShot);
          }
 
-         private Quaternion GenerateSpreadAngle()
+         private Quaternion GenerateSpreadAngle(float timeSinceFiring)
          {
-             var spread = _weaponProperties.BulletSpread;
+             var spreadRate = GetBulletSpreadRate(timeSinceFiring);
+             var spread = _weaponProperties.BulletSpread * spreadRate;
```

Comment:
This is more of what i had in mind
```suggestion
             private float GetSpreadRate(float timeSinceFiring) {
             var spreadRate = GetBulletSpreadRate(timeSinceFiring);
             return  _weaponProperties.BulletSpread * spreadRate;
             }
             
             var spread = GetSpreadRate(0.5)
             
             private Quaternion GenerateSpreadAngle(float spread) {
             
             }
```